### PR TITLE
Hide control bar buttons based on Rooms role permissions

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Hide control bar buttons based on Rooms role permissions instead of disabling them. These were changes requested by Alex Pereira before public preview.

Will add ControlBar unit tests after. It is surprisingly challenging.

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2972689
https://skype.visualstudio.com/SPOOL/_workitems/edit/2972694
https://skype.visualstudio.com/SPOOL/_workitems/edit/2972697
https://skype.visualstudio.com/SPOOL/_workitems/edit/2972703
https://skype.visualstudio.com/SPOOL/_workitems/edit/2974319

# How Tested

Ran local sample

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by PorterNan — APPROVED** [review]
Thanks for making this change in such a velocity! There are a concern around here I wanna discuss:
1. Since we are moving rendering responsibility from component itself to composite, we removed a bunch of "disable" code - since we don't provide support for component level, this is fine. When we support component level things - what if a developer still render a component under an unsupported role? 

My answer is still, we need to gray the component out since we are going to have role info from stateful client and selector, it should not be a concern for a long run(we will disable these component in selector) - but maybe we can consider reserve these gray code out code until then?

Would like to hear voices from @alkwa-msft @prabhjot-msft as well

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 1]
```diff
@@ -60,7 +62,7 @@ export type CallControlsProps = {
 const controlBarStyles = memoizeFunction((background: string) => ({ root: { background: background } }));
 
 /**
- * @private
+ * @public
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:65]: CallControl isn't an exposed api, why do we have a jsdoc change here?

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 2]
```diff
@@ -38,7 +40,7 @@ import { useAdapter } from '../adapter/CallAdapterProvider';
 import { isDisabled } from '../utils';
 
 /**
- * @private
+ * @public
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:43]: Same here, revert back to private

**Review by PorterNan — COMMENTED** [review]
Looks like we still need remove ellipsis button from people pane - is this covered in another work item?

**Comment by mgamis-msft** [comment]
> Looks like we still need remove ellipsis button from people pane - is this covered in another work item?



> Thanks for making this change in such a velocity! There are a concern around here I wanna discuss:
> 
> 1. Since we are moving rendering responsibility from component itself to composite, we removed a bunch of "disable" code - since we don't provide support for component level, this is fine. When we support component level things - what if a developer still render a component under an unsupported role?
> 
> My answer is still, we need to gray the component out since we are going to have role info from stateful client and selector, it should not be a concern for a long run(we will disable these component in selector) - but maybe we can consider reserve these gray code out code until then?
> 
> Would like to hear voices from @alkwa-msft @prabhjot-msft as well

It's easy enough to keep the components grayed out by keeping them hooked to the Permissions context. Interested in what @alkwa-msft @prabhjot-msft think as well.

**Review by prabhjot-msft — APPROVED** [review]
Looks good to me! I agree to Nan's point even if we are hiding the components, we should still considering to disable components based on Permissions if user decides to use components instead of composites.

**Pull request merged**